src/lib/features/export-import-toggles/export-import-service.ts

```diff
@@ -252,7 +253,7 @@ export default class ExportImportService
         user: IUser,
         mode = 'regular' as Mode,
     ): Promise<void> {
-        await Promise.all([
+        await allSettledWithRejection([
             this.verifyStrategies(dto),
             this.verifyContextFields(dto),
             this.importPermissionsService.verifyPermissions(dto, user, mode),
```

Comment:
verify permissions was always running outside of the current transaction when segments checked failed. Segment check was one simple DB query whole permission check involved multiple queries.